robocop/checkers/misc.py

```diff
@@ -525,15 +564,47 @@ def check_adjacent_ifs(self, node):
                 previous_if = None
         self.generic_visit(node)
 
-    @staticmethod
-    def compare_conditions(if_node, other_if_node):
+    def compare_conditions(self, if_node, other_if_node):
+        if not self.compare_assign(if_node, other_if_node):
+            return False
         while if_node is not None and other_if_node is not None:
             if if_node.condition != other_if_node.condition:
                 return False
             if_node = if_node.orelse
             other_if_node = other_if_node.orelse
         return if_node is None and other_if_node is None
 
+    @staticmethod
+    def normalize_var_name(name):
+        return name.lower().replace("_", "").replace(" ", "").replace("=", "")
+
+    def compare_assign(self, if_node, other_if_node):
```

Comment:
Maybe lest change the method name to `compare_assign_tokens`? `compare_assign` sounds strange.